extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/caffeine/CaffeineCacheInfoBuilder.java

```diff
@@ -1,30 +1,63 @@
 package io.quarkus.cache.runtime.caffeine;
 
+import static io.quarkus.cache.runtime.CacheConfig.CaffeineConfig.CaffeineCacheConfig;
+
 import java.util.Collections;
 import java.util.Set;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 
 import io.quarkus.cache.runtime.CacheConfig;
-import io.quarkus.cache.runtime.CacheConfig.CaffeineConfig.CaffeineNamespaceConfig;
 
 public class CaffeineCacheInfoBuilder {
 
     public static Set<CaffeineCacheInfo> build(Set<String> cacheNames, CacheConfig cacheConfig) {
         if (cacheNames.isEmpty()) {
             return Collections.emptySet();
         } else {
-            return cacheNames.stream().map(cacheName -> {
-                CaffeineCacheInfo cacheInfo = new CaffeineCacheInfo();
-                cacheInfo.name = cacheName;
-                CaffeineNamespaceConfig namespaceConfig = cacheConfig.caffeine.namespace.get(cacheInfo.name);
-                if (namespaceConfig != null) {
-                    namespaceConfig.initialCapacity.ifPresent(capacity -> cacheInfo.initialCapacity = capacity);
-                    namespaceConfig.maximumSize.ifPresent(size -> cacheInfo.maximumSize = size);
-                    namespaceConfig.expireAfterWrite.ifPresent(delay -> cacheInfo.expireAfterWrite = delay);
-                    namespaceConfig.expireAfterAccess.ifPresent(delay -> cacheInfo.expireAfterAccess = delay);
-                    cacheInfo.metricsEnabled = namespaceConfig.metricsEnabled;
+            CaffeineCacheConfig defaultConfig = cacheConfig.caffeine.defaultConfig;
+
+            return cacheNames.stream().map(new Function<String, CaffeineCacheInfo>() {
```

Comment:
No lambdas in runtime code please :)

Reply:
Is there a lambda in that line?

Reply:
Right, I should have been more precise :). 

Let's try to avoid streams when we can - in this case it should be trivial.

Reply:
Oh okay, I can definitely get rid of that stream!